os/kernel/binary_manager/binary_manager_binfile.c

```diff
@@ -222,12 +227,14 @@ int binary_manager_scan_ubin(int bin_idx)
  ****************************************************************************/
 int binary_manager_create_entry(int requester_pid, char *bin_name, int version)
 {
+#ifdef CONFIG_APP_BINARY_SEPARATION
```

Comment:
This file is built only when `CONFIG_BINARY_MANAGER` is enabled. 
And `CONFIG_BINARY_MANAGER` has dependency with `CONFIG_APP_BINARY_SEPARATION`.
You'd better check the dependency of all configs related to binary manager and app separation.